client/src/main/scala/org/apache/celeborn/client/ChangePartitionManager.scala

```diff
@@ -151,36 +170,22 @@ class ChangePartitionManager(
       oldPartition,
       cause)
 
-    // If new slot for the partition has been allocated, reply and return.
-    // Else register and allocate for it.
-    getLatestPartition(shuffleId, partitionId, oldEpoch).foreach { latestLoc =>
-      context.reply(
-        partitionId,
-        StatusCode.SUCCESS,
-        Some(latestLoc),
-        lifecycleManager.workerStatusTracker.workerAvailable(oldPartition))
-      logDebug(s"New partition found, old partition $partitionId-$oldEpoch return it." +
-        s" shuffleId: $shuffleId $latestLoc")
-      return
-    }
-
-    requests.synchronized {
+    val locksForShuffle = locks.computeIfAbsent(shuffleId, locksRegisterFunc)
```

Comment:
With the introduction of fine-grained locking, request locks may no longer be an issue, and [CELEBORN-1509](https://github.com/apache/celeborn/pull/2627) might not be necessary. Consider this scenario: partition 0 has a latest epoch of 4, and the revive partition requests are 3, 3, 3, 4, 3, 3, 3.

After [CELEBORN-1509](https://github.com/apache/celeborn/pull/2627), the requests for epoch 3 will receive the latest epoch 4. Even after updating the request to epoch 4, the remaining requests will still return epoch 4. Therefore, I believe it is better to check the unhandled partition IDs first. Without [CELEBORN-1509](https://github.com/apache/celeborn/pull/2627), the initial requests for epoch 3 will also receive epoch 4. However, once epoch 4 is handled, the remaining requests will wait for a new partition epoch.  

cc @waitinfuture  @zhaostu4